From b189efd8bd9b96910cbca488100c4b41e1cac4b5 Mon Sep 17 00:00:00 2001
From: Stephan Linz <linz@li-pro.net>
Date: Mon, 15 Apr 2013 16:53:31 +0200
Subject: [PATCH] backported from trunk

Signed-off-by: Stephan Linz <linz@li-pro.net>
---
 Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index 7ce0947..fd61a56 100644
--- a/Makefile
+++ b/Makefile
@@ -50,7 +50,7 @@ export	ARCH CPU BOARD VENDOR SOC
 include $(TOPDIR)/config.mk
 
 ifndef CROSS_COMPILE
-ifeq ($(HOSTARCH),ppc)
+ifeq ($(HOSTARCH),$(ARCH))
 CROSS_COMPILE =
 else
 ifeq ($(ARCH),ppc)
-- 
1.7.1

